### DISCUSSION TITLE
Migrating from using --out-file via CLI in Parcel V2

### DISCUSSION THREAD

**Discussion by lqze** [parcel-bundler/parcel#5552] [Q&A]
Hi there,

I'm struggling to understand the correct replacement for `--out-file`, particularly if you are not simply bundling one `js` file. 

How can I go from `parcel build index.js --out-file index.min.js && parcel build index.scss --out-file index.min.css` from Parcel V1 to Parcel V2.

Thanks for the help.

**Comment by valentinbdv** [comment]
Hi, I am having the same issue and I don't know what would be the best replacement.

Thanks for your help.

**Reply by lqze** [reply]
Yeah I just went back to webpack. Too much hassle with Parcel 2.

**Reply by euclid1990** [reply]
@valentinbdv May be I also come back to Webpack =))

**Comment by DeMoorJasper** [comment]
Maybe this could help: https://v2.parceljs.org/configuration/package-json/#targets

**Reply by valentinbdv** [reply]
It did!
I didn't know how target was working but after some test, it makes the job.
I think you could add that solution on the migration page: https://v2.parceljs.org/getting-started/migration/
Because it seems like you can only use the default key in package.json which doesn't really allow to use the export file name depending on the script.
Thanks a lot for the suggestion!

**Reply by Maker-Mark** [reply]
I dont see how this would let you set a custom output file. Can you provide an example?

**Reply by iamxeeshankhan** [reply]
Hi @Maker-Mark 
Did you find the answer to your question? I'm facing the custom out file problem with parcel's latest version. As you know we cannot use --out-file anymore so if you now know a replacement to this, then please share it.

**Reply by Maker-Mark** [reply]
Nope!

**Reply by avoguga** [reply]
For those who still come here, and maybe even me in the future, I managed to do the same as the --out-file with the orientation of the docs. 

https://parceljs.org/features/targets/#multiple-targets

Here a package.json example:

```
{
  "renameFile": "dist/renamedFile.min.js",
  "targets": {
    "renameHere": {},
  },
  "scripts": {
      "build:web": "parcel build src/index.tsx --target renamedFile",
  }
}
```

**Comment by ttodua** [comment]
I made it work by below:
```
{
  ...
  "main": "dist/desired_name.js",  // the output file name, as it is the final entry for anything other
  "scripts": {
    "parcel": "parcel build",
  },
  "__comment__": "below targets key if for PARCEL CONFIG",
  "my_server": "dist/desired_name.js",
  "targets": {
    "my_server": {
      "context": "node",
      "source": "./src/index.ts",
      "distDir": "./dist",
      "engines": {
        "node": ">=20.x"
      },
      "includeNodeModules": false,
      "optimize": true,
      "sourceMap": false,
      "outputFormat": "commonjs"
    }
  }
}
```
